src/Sentry.Unity/SentryInitialization.cs

```diff
@@ -51,147 +28,14 @@ public static void Init()
-                    SentryUnityCompression.Auto => CompressionLevel.NoCompression,
-                    SentryUnityCompression.NoCompression => CompressionLevel.NoCompression,
-                    _ => CompressionLevel.NoCompression
-                };
-                o.AddEventProcessor(new UnityEventProcessor());
-                o.AddExceptionProcessor(new UnityEventExceptionProcessor());
-            });
-
-            // TODO: Consider ensuring this code path doesn't require UI thread
-            // Then use logMessageReceivedThreaded instead
-            void OnApplicationOnLogMessageReceived(string condition, string stackTrace, LogType type) => OnLogMessageReceived(condition, stackTrace, type, options);
-
-            Application.logMessageReceived += OnApplicationOnLogMessageReceived;
-            Application.quitting += () =>
-            {
-                // Note: iOS applications are usually suspended and do not quit. You should tick "Exit on Suspend" in Player settings for iOS builds to cause the game to quit and not suspend, otherwise you may not see this call.
-                //   If "Exit on Suspend" is not ticked then you will see calls to OnApplicationPause instead.
-                // Note: On Windows Store Apps and Windows Phone 8.1 there is no application quit event. Consider using OnApplicationFocus event when focusStatus equals false.
-                // Note: On WebGL it is not possible to implement OnApplicationQuit due to nature of the browser tabs closing.
-                Application.logMessageReceived -= OnApplicationOnLogMessageReceived;
-                SentrySdk.Close();
-            };
-
-            IDictionary<string, string>? data = null;
-            if (SceneManager.GetActiveScene().name is { } name)
-            {
-                data = new Dictionary<string, string> { { "scene", name } };
-            }
-            SentrySdk.AddBreadcrumb("BeforeSceneLoad", data: data);
-
-            options.Logger?.Log(SentryLevel.Debug, "Complete Sentry SDK initialization.");
-
-            IsInit = true;
-        }
-
-        // Happens with Domain Reloading
-        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
-        private static void SubsystemRegistration() => SentrySdk.AddBreadcrumb("SubsystemRegistration");
-
-        private static void OnLogMessageReceived(string condition, string stackTrace, LogType type, UnitySentryOptions options)
-        {
-            // TODO: 'options' not used yet
-            _ = options;
-
-            var debounced = type switch
-            {
-                LogType.Error or LogType.Exception or LogType.Assert => ErrorTimeDebounce.Debounced(),
-                LogType.Log => LogTimeDebounce.Debounced(),
-                LogType.Warning => WarningTimeDebounce.Debounced(),
-                _ => true
-            };
-            if (!debounced)
-            {
-                return;
-            }
+            SentryUnity.Init(options);

```

Comment:
We'll need an opt-out I guess. So probably need to read the SentryOptions json to check for a flag to decide whether to init here or if the user is going to do it programatically.

Reply:
Let's discuss.